packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -1,124 +1,120 @@
+      setDateType('futureDate');
+    } else {
+      setDateType('today');
+    }
+  }, [date]);
 
-  const filteredAppointments = appointmentServiceType
-    ? appointmentList.filter(({ serviceTypeUuid }) => serviceTypeUuid === appointmentServiceType)
-    : appointmentList;
-  const rowData = appointmentList.map((appointment, index) => {
-    return {
-      id: `${index}`,
-      ...appointment,
-    };
-  });
-  const filteredRow = appointmentServiceType
-    ? rowData.filter((app) => app.serviceTypeUuid === appointmentServiceType)
-    : rowData;
+  // set the default tab on first render
+  useEffect(() => {
+    if (!currentTab) {
+      setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
+    }
+  }, [scheduledAppointmentPanels]);
 
-  const appointmentsBaseTableConfig = {
-    Scheduled: {
-      appointments: filteredAppointments,
-      isLoading,
-      tableHeading: t('scheduled', 'Scheduled'),
-      visits,
-      scheduleType,
-    },
-    CameEarly: {
-      appointments: earlyAppointmentList,
-      isLoading: loading,
-      tableHeading: t('cameEarly', 'Came early'),
-      visits,
-      scheduleType,
-    },
-    Honoured: {
-      appointments: filteredRow,
-      isLoading,
-      tableHeading: t('honored', 'Honored'),
-      visits,
-      scheduleType,
-    },
-    Rescheduled: {
-      appointments: filteredRow,
-      isLoading,
-      tableHeading: t('rescheduled', 'Rescheduled'),
-      visits,
-      scheduleType,
-    },
-    Pending: {
-      appointments: filteredRow,
-      isLoading,
-      tableHeading: isDateInPast ? t('missed', 'Missed') : t('notArrived', 'Not arrived'),
-      visits,
-      scheduleType,
-    },
-  };
+  // set the default tab whenever the date type changes
+  useEffect(() => {
+    setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
+  }, [dateType]);
 
-  const currentConfig = appointmentsBaseTableConfig[scheduleType];
+  const getDefaultTab = (panels) => {
+    return panels?.find(
+      (panel) =>
+        (isDateInPast && panel.meta.showForPastDate) ||
+        (isToday && panel.meta.showForToday) ||
+        (isDateInFuture && panel.meta.showForFutureDate),
+    )?.meta.title;
+  };
 
   return (
     <>
-      {isToday && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setScheduleType(name)}>
-          <Switch name={'Scheduled'} text={t('scheduled', 'Scheduled')} />
-          <Switch name={'Honoured'} text={t('honored', 'Honored')} />
-          <Switch name={'Pending'} text={t('notArrived', 'Not arrived')} />
-          <Switch name={'CameEarly'} text={t('cameEarly', 'Came early')} />
+      {dateType === 'today' && (
+        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
+          {scheduledAppointmentPanels.map((panel) => {
+            return panel.meta.showForToday ? (
+              <Switch name={panel.meta.title} text={t(panel.meta.title, panel.meta.title)} />
```

Comment:
When calling `map()` to render an array of React components, it's best to provide a key for them:

```suggestion
          {scheduledAppointmentPanels.map((panel, i) => {
            return panel.meta.showForToday ? (
              <Switch key={i} name={panel.meta.title} text={t(panel.meta.title, panel.meta.title)} />
```

This may actually be a counter-argument to my "We don't need a fragment" claim:

```ts
          {scheduledAppointmentPanels.map((panel) => {
            return panel.meta.showForToday ? (
              <Switch key={panel.meta.title} name={panel.meta.title} text={t(panel.meta.title, panel.meta.title)} />
            ) : (
              <Fragment key={panel.meta.title} />
            )
```

Also using `panel.meta.title` doesn't assume that the `scheduleAppointmentPanels` are always returned in the same order.

Reply:
Makes sense about the key @ibacher , will add it, but are you saying that even with a key we can't guarantee order?  how does adding a fragment help us out, does it help mantain order even for elements that aren't displayed?  Thanks!